### PR TITLE
Convert Button and UIAction to TypeScript

### PR DESCRIPTION
**Pull request by lavelle**

### CODE REVIEW

**Comment by lavelle** [comment]
gatsby failing with an unrelated CMS thing. Everything looks good when I run the docs locally, including `iconRight` which accidentally got deleted but I then later fixe.d

File: packages/thumbprint-react/components/UIAction/get-button-props.ts [hunk 1]
```diff
@@ -1,3 +1,13 @@
+interface ButtonProps {
+    type: 'button' | 'submit';
+    onClick?: () => void;
```
Comment (danoc) [inline packages/thumbprint-react/components/UIAction/get-button-props.ts]: Are these not supposed to include the event as a parameter? He's `onClick`, for example:

```
(event: React.MouseEvent<HTMLButtonElement, MouseEvent>) => void
```

File: packages/thumbprint-react/components/UIAction/test.tsx [hunk 2]
```diff
@@ -242,38 +242,20 @@ describe('Plain', () => {
...
-    test('left and right icons can be used together', () => {
-        const wrapper = mount(
-            <Plain iconLeft={<svg>Swan</svg>} iconRight={<svg>Duck</svg>}>
-                Goose
-            </Plain>,
-        );
-        expect(wrapper.find('svg')).toHaveLength(2);
-        expect(wrapper.text()).toBe('SwanGooseDuck');
-        expect(wrapper).toMatchSnapshot();
-    });
```
Comment (danoc) [inline packages/thumbprint-react/components/UIAction/test.tsx]: Why do these two go away?

File: packages/thumbprint-react/components/UIAction/get-button-props.ts (hunk 1 above)
Comment (lavelle) [reply packages/thumbprint-react/components/UIAction/get-button-props.ts]: yes, i wrote this code before i knew the react dom types so well. fixed here and elsewhere.

File: packages/thumbprint-react/components/UIAction/test.tsx (hunk 2 above)
Comment (lavelle) [reply packages/thumbprint-react/components/UIAction/test.tsx]: oops, that's what you get for cherry picking a commit from 3 months ago. good catch.

**Pull request merged**